src/tasks/batch.ts

```diff
@@ -121,94 +126,69 @@ export class BatchCommitter {
                 commitment.approved ? 'Approved' : 'Disapproved',
             ]
             const encodedArgs: Uint8Array[] = encodeStringArgs(this.contractApi.abi, fragment, args)
-            const extrinsic = await this.contractApi.buildExtrinsic('dappUserCommit', encodedArgs)
-
-            txs.push(extrinsic)
+            const { extrinsic, options } = await this.contractApi.buildExtrinsic('dappUserCommit', encodedArgs)
+            totalRefTime = totalRefTime.add(
+                this.contractApi.api.registry.createType('WeightV2', options.gasLimit).refTime.toBn()
+            )
+            // Check if we have a maximum number of transactions that we can successfully submit in a block
+            if (
+                totalRefTime.mul(BN_TEN_THOUSAND).div(maxBlockWeight.refTime.toBn()).toNumber() / 100 >
+                this.batchCommitConfig.maxBatchExtrinsicPercentage
+            ) {
+                breakBatchLoop = true
```

Comment:
you can put `break` here instead of using the `breakBatchLoop` variable

Reply:
thx, have modified and tests running.